presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushPartialAggregationThroughExchange.java

```diff
@@ -94,7 +95,7 @@ public Result apply(AggregationNode aggregationNode, Captures captures, Context
             return Result.ofPlanNode(split(aggregationNode, context));
         }
 
-        if (!decomposable) {
+        if (!decomposable || !SystemSessionProperties.preferPartialAggregation(context.getSession())) {
```

Comment:
Static import the function name since the name is readable by itself

Reply:
I saw both versions almost equally used in code though.